sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/projects/aio/_operations/_operations.py

```diff
@@ -0,0 +1,1827 @@
+                path_format_arguments = {
+                    "Endpoint": self._serialize.url("self._config.endpoint", self._config.endpoint, 'str', skip_quote=True),
+                }
+                request.url = self._client.format_url(request.url, **path_format_arguments)
+
+            else:
+                
+                request = build_list_synonyms_request(
+                    project_name=project_name,
+                    api_version=api_version,
+                    top=top,
+                    skip=skip,
+                    maxpagesize=maxpagesize,
+                )
+                path_format_arguments = {
+                    "Endpoint": self._serialize.url("self._config.endpoint", self._config.endpoint, 'str', skip_quote=True),
+                }
+                request.url = self._client.format_url(next_link, **path_format_arguments)
+
+                path_format_arguments = {
+                    "Endpoint": self._serialize.url("self._config.endpoint", self._config.endpoint, 'str', skip_quote=True),
+                }
+                request.method = "GET"
+            return request
+
+        async def extract_data(pipeline_response):
+            deserialized = _loads(pipeline_response.http_response.body())
+            list_of_elem = deserialized["value"]
+            if cls:
+                list_of_elem = cls(list_of_elem)
+            return deserialized.get("nextLink", None), AsyncList(list_of_elem)
+
+        async def get_next(next_link=None):
+            request = prepare_request(next_link)
+
+            pipeline_response = await self._client._pipeline.run(request, stream=False, **kwargs)
+            response = pipeline_response.http_response
+
+            if response.status_code not in [200]:
+                map_error(status_code=response.status_code, response=response, error_map=error_map)
+                raise HttpResponseError(response=response)
+
+            return pipeline_response
+
+
+        return AsyncItemPaged(
+            get_next, extract_data
+        )
+    list_synonyms.metadata = {'url': '/query-knowledgebases/projects/{projectName}/synonyms'}  # type: ignore
+
+    @distributed_trace_async
+    async def update_synonyms(
+        self,
+        project_name: str,
+        synonyms: JSONType,
+        **kwargs: Any
+    ) -> None:
+        """Updates all the synonyms of a project.
+
+        Updates all the synonyms of a project.
+
+        :param project_name: The name of the project to use.
+        :type project_name: str
+        :param synonyms: All the synonyms of a project.
+        :type synonyms: JSONType
+        :return: None
+        :rtype: None
+        :raises: ~azure.core.exceptions.HttpResponseError
+
+        Example:
+            .. code-block:: python
+
+                # JSON input template you can fill out and use as your body input.
+                synonyms = {
+                    "nextLink": "str",  # Optional.
+                    "value": [
```

Comment:
Do we have a test/sample for `update_synonyms`?
This sample input block looks incorrect, it seems to be the output of the paging operation... presumably that's not the expected input structure?

Reply:
this is the same request body as in the swagger [here](https://github.com/Azure/azure-rest-api-specs/blob/219b2b3dea0233e3cb43f72d2764c0e4e4d8fcd3/specification/cognitiveservices/data-plane/Language/stable/2021-10-01/questionanswering-authoring.json#L646)
and [here](https://github.com/Azure/azure-rest-api-specs/blob/219b2b3dea0233e3cb43f72d2764c0e4e4d8fcd3/specification/cognitiveservices/data-plane/Language/stable/2021-10-01/questionanswering-authoring.json#L1263)